### PR TITLE
EPMRPP-83221 || update the links on project settings pages connected to data privacy

### PR DESCRIPTION
**Pull request by Vadim73i**

### CODE REVIEW

File: app/src/layouts/settingsLayout/settingsLayout.jsx [hunk 1]
```diff
@@ -15,26 +15,130 @@
...
+      </a>
+    ),
+  ];
+
+  return (
+    <div className={cx('container')}>
+      {navigation && <div className={cx('navigation')}>{navigation}</div>}
+      <div className={cx('section')}>
+        <div className={cx('section-content')}>{children}</div>
+        <div className={cx('footer')}>
```
Comment (AmsterGet) [inline app/src/layouts/settingsLayout/settingsLayout.jsx:102]: Please move this Layout's footer to its own component.

File: app/src/layouts/settingsLayout/settingsLayout.jsx [hunk 2]
```diff
@@ -15,26 +15,130 @@
...
+    </a>,
+    <a href={referenceDictionary.rpEmail} target="_blank" className={cx('link')}>
+      <span>{formatMessage(messages.contactUs)}</span>
+    </a>,
+    <a href={referenceDictionary.rpEpam} target="_blank" className={cx('link')}>
+      <span>{formatMessage(messages.epam)}</span>
+    </a>,
+    <a href={referenceDictionary.rpDoc} target="_blank" className={cx('link')}>
+      <span>{formatMessage(messages.documentation)}</span>
+    </a>,
```
Comment (AmsterGet) [inline app/src/layouts/settingsLayout/settingsLayout.jsx:89]: You can build that elements during config mapping and left only messages and links in the config.

File: app/src/layouts/settingsLayout/settingsLayout.jsx [hunk 3]
```diff
@@ -15,26 +15,130 @@
...
+    id: 'SettingsLayout.documentation',
+    defaultMessage: 'Documentation',
+  },
+  privacyPolicy: {
+    id: 'SettingsLayout.privacyPolicy',
+    defaultMessage: 'Privacy Policy',
+  },
+  rights: {
+    id: 'SettingsLayout.rights',
+    defaultMessage: '© Report Portal 2023 All rights reserved',
```
Comment (AmsterGet) [inline app/src/layouts/settingsLayout/settingsLayout.jsx:56]: Do not hardcode the actual year value.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/layouts/settingsLayout/footer/footer.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+FooterBase.propTypes = {
+  buildVersion: PropTypes.string.isRequired,
+  instanceType: PropTypes.string.isRequired,
+  date: PropTypes.instanceOf(Date),
+};
+FooterBase.defaultProps = {
+  date: new Date(),
+};
+export const Footer = connect((state) => ({
```
Comment (Bam6ycha) [inline app/src/layouts/settingsLayout/footer/footer.jsx:141]: Clould you explain, why we are using `connect` here?
My suggestion is to rewrite it to hooks.
Connect works differently from useSelector, but in this case it does not metter.

File: app/src/layouts/settingsLayout/settingsLayout.jsx [hunk 5]
```diff
@@ -15,130 +15,30 @@
...
+  <div className={cx('container')}>
+    {navigation && <div className={cx('navigation')}>{navigation}</div>}
+    <div className={cx('section')}>
+      <div className={cx('section-content')}>{children}</div>
+      <Footer />
     </div>
-  );
-};
+  </div>
+);
```
Comment (Bam6ycha) [inline app/src/layouts/settingsLayout/settingsLayout.jsx:34]: Remove empty line.

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

File: app/src/layouts/settingsLayout/footer/footer.jsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  build: {
+    id: 'SettingsLayout.footer.build',
+    defaultMessage: 'Build: {buildVersion}',
+  },
+});
+
+const FooterBase = ({ buildVersion, instanceType, date }) => {
+  const { formatMessage } = useIntl();
+
+  const currentYear = date.getFullYear();
```
Comment (AmsterGet) [inline app/src/layouts/settingsLayout/footer/footer.jsx:67]: It looks like the date is not actually a prop and is never passed through props.
Why not calculate the actual year right here?

```suggestion
  const currentYear = (new Date()).getFullYear();
```

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Review by AmsterGet — APPROVED** [review]

**Review by Bam6ycha — APPROVED** [review]

**Pull request merged**